ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs

```diff
@@ -368,15 +368,19 @@ protocolInfoShelley genesis protVer mbCredentials =
         -- during the previous epoch. We create a "fake" snapshot in order to
         -- establish an initial stake distribution.
         initSnapShot = SL.SnapShot
-          { SL._stake = SL.Stake . Map.mapKeys addrKeyHash $ sgInitialFunds genesis
+          { SL._stake = SL.Stake . Map.fromList $
+              [ (stakeCred, stake)
+              | (addr, stake) <- Map.toList (sgInitialFunds genesis)
+              , let Just stakeCred = addrStakeCred addr
```

Comment:
You mean to do filtering here, not an irrefutable pattern match.
```suggestion
              , Just stakeCred <- [addrStakeCred addr]
```
This is a reasonable trick for this situation.